### PR TITLE
chore: update Go version

### PR DESCRIPTION
**Pull request by matejvasek**

### CODE REVIEW

**Comment by matejvasek** [comment]
@dsimansk  @mvinkler PTAL

File: ci-operator/config/openshift-knative/kn-plugin-func/openshift-knative-kn-plugin-func-release-next.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ build_root:
   image_stream_tag:
     name: builder
     namespace: ocp
-    tag: rhel-8-golang-1.20-openshift-4.14
+    tag: rhel-8-release-golang-1.21-openshift-4.16
```
Comment (jrangelramos) [inline ci-operator/config/openshift-knative/kn-plugin-func/openshift-knative-kn-plugin-func-release-next.yaml:5]: Hi @matejvasek reharsal failed.. maybe try `rhel-8-release-golang-1.21-openshift-4.16` instead

**Comment by matejvasek** [comment]
PTAL @jrangelramos

**Comment by jrangelramos** [comment]
/pj-rehearse pull-ci-openshift-knative-kn-plugin-func-release-next-unit

**Comment by jrangelramos** [comment]
Hello @dsimansk do you know why retest command is not working here? I wanna retest against the latest commit

**Pull request merged**